lpd/tests/test_views.py

```diff
@@ -331,14 +385,63 @@ def test_post_qualitative_answers(self, patched_send_learner_data, patched_calcu
         # Make sure correct set of scores was passed to method for sending learner data to adaptive engine
         patched_send_learner_data.assert_called_once_with(self.user, list(scores))
 
+        # Make sure submission data was updated
+        self._assert_submission_data()
+
+    @patch('lpd.views.LPDSubmitView._process_quantitative_answers', new=MagicMock(return_value=[]))
+    def test_post_qual_answers_split_answers(self):
+        """
+        Test that `post` correctly processes qualitative answers.
+        """
+        question1 = QualitativeQuestionFactory(split_answer=False)
+        question2 = QualitativeQuestionFactory(split_answer=True)
+
+        qualitative_answers = [
+            {
+                'question_id': 1,
+                'answer_text': 'This is a very clever answer, I must say.',
+            },
+            {
+                'question_id': 2,
+                'answer_text': 'This is not a very clever answer, but an answer nonetheless.',
+            }
+        ]
+        self.data['qualitative_answers'] = json.dumps(qualitative_answers)
+
+        response = self.client.post(reverse('lpd:submit'), self.data)
+        self.assertEqual(response.status_code, 200)
+
+        qualitative_answers = QualitativeAnswer.objects.all()
+        self.assertEqual(qualitative_answers.count(), 3)
+
+        # Check answers individually
+        question1_answers = qualitative_answers.filter(question=question1)
+        self.assertEqual(question1_answers.count(), 1)
+
+        question1_answer = question1_answers.get()
+        self.assertEqual(question1_answer.learner, self.user)
+        self.assertEqual(question1_answer.text, 'This is a very clever answer, I must say.')
+
+        question2_answers = qualitative_answers.filter(question=question2)
+        self.assertEqual(question2_answers.count(), 2)
+
+        for question2_answer, expected_text in zip(
+                question2_answers, ['This is not a very clever answer', 'but an answer nonetheless.']
```

Comment:
@itsjeyd Are we guaranteed that `question2_answers` will always have this order? (I mean: `['This is not a very clever answer', 'but an answer nonetheless.']`, not: `['but an answer nonetheless.', 'This is not a very clever answer']`)

Reply:
@tomaszgy You're right, we're not :)

Since we [`delete`](https://github.com/open-craft/learner-profile-dashboard/blob/7d4f5eabd67af114f4445f0d7e88f8e739cf184b/lpd/views.py#L144-L145) all answers belonging to a qualitative question and then [`create`](https://github.com/open-craft/learner-profile-dashboard/blob/7d4f5eabd67af114f4445f0d7e88f8e739cf184b/lpd/views.py#L151-L156) a new answer for each answer component, I kind of glossed over this and assumed that Django would Do The Right Thing&trade; and always return the answers in the same order. But that's not the case, the order actually [depends on the underlying RDBMS](https://stackoverflow.com/a/7164126/1199226).

However, this isn't too big of a deal for HGSE's use case: The only question that they want to use the `split_answer` feature for asks learners for a comma-separated list of words, e.g. "blue, green, yellow". If the LPD displayed that back to the learner as "green, yellow, blue" upon subsequent page loads, they might wonder why, but that wouldn't change the fact that their answer would still be intact :)

In any case, in 8a45320e0fbad80e5dad40a0bf10c893482c1a02 I changed the`get_answer` logic to order individual answers belonging to a question by `id`, so the order in which the LPD displays them back to the learner should always match the order in which they were created/listed by the learner.

Reply:
@itsjeyd cool, thanks for explaining that and the update :)